src/vstart.sh

```diff
@@ -349,14 +349,18 @@ if [ "$overwrite_conf" -eq 0 ]; then
 else
     if [ "$new" -ne 0 ]; then
         # only delete if -n
```

Comment:
So the purpose of this `if [ "$new" -ne 0 ]` section is to cleanup remnants (asok dir) from the previous vstart run. I think it makes sense to try to remove asok_dir only if $conf_fn exists, otherwise (when ceph-conf is called without `-c $conf_fn` we may delete someone other dir specified in the global ceph.conf.

Thus I think you should check if $conf_fn exists end skip the removal if it does not. I.e. the condition could be:
```
     if [ "$new" -ne 0 -a -e "$conf_fn" ]; then
```